### PR TITLE
Rel/6.1.0 preview1

### PR DESCRIPTION
**Pull request by simeoncran**
Toolkit version 6.1 preview.

### CODE REVIEW

File: VERSION_HISTORY.md [hunk 1]
```diff
@@ -2,6 +2,64 @@
...
+## 6.1.0
+This release brings a new theme property binding feature, better Lottie feature support, performance improvements and bug fixes. 
+
+**Theme property binding**
+
+This feature enables properties in Lottie files to be bound to values at runtime. This enables scenarios such as synchronizing colors in your Lottie file to the Windows theme colors.
+
+Theme property binding was developed to support fill color binding, however it also supports StrokeWidth and opens the door to enabling other property bindings.
+
+To make an After Effects color fill bound to a property at runtime, the name of the fill needs to include (css-like) syntax such as the following: "{color:var( Foreground)}". This will create a property in the generated code called "Foreground".
```
Comment (derobinsMSFT) [inline VERSION_HISTORY.md:14]: >css [](start = 107, length = 3)

All caps?

File: VERSION_HISTORY.md [hunk 2]
```diff
@@ -2,6 +2,64 @@
 
 Lottie-Windows aligns its major version with the Windows Community Toolkit.
 
+## 6.1.0
+This release brings a new theme property binding feature, better Lottie feature support, performance improvements and bug fixes.
```
Comment (derobinsMSFT) [inline VERSION_HISTORY.md:6]: >  [](start = 128, length = 1)

Nit: several lines added here have extra spaces at the end.

File: VERSION_HISTORY.md [hunk 3]
```diff
@@ -2,6 +2,64 @@
...
+
+This feature enables properties in Lottie files to be bound to values at runtime. This enables scenarios such as synchronizing colors in your Lottie file to the Windows theme colors.
+
+Theme property binding was developed to support fill color binding, however it also supports StrokeWidth and opens the door to enabling other property bindings.
+
+To make an After Effects color fill bound to a property at runtime, the name of the fill needs to include (css-like) syntax such as the following: "{color:var( Foreground)}". This will create a property in the generated code called "Foreground". 
+
+Property binding default values are exposed as constants. This allows these default values to be used in supporting code without having to instantiate the IAnimatedVisualSource object. 
+
+The method ColorAsVector4(…) method in generated code is to help developers set color properties. Due to a limitation of the Windows.UI.Composition APIs, we can't use the Windows.UI.Xaml.Color type for doing property binding. We use Vector4 instead and this method ensures the conversion from color to Vector4 is done the same way by developers as is expected by generated code.
```
Comment (derobinsMSFT) [inline VERSION_HISTORY.md:18]: >method [](start = 4, length = 6)

Redundant-- remove.

File: VERSION_HISTORY.md [hunk 4]
```diff
@@ -2,6 +2,64 @@
...
+Property binding default values are exposed as constants. This allows these default values to be used in supporting code without having to instantiate the IAnimatedVisualSource object. 
+
+The method ColorAsVector4(…) method in generated code is to help developers set color properties. Due to a limitation of the Windows.UI.Composition APIs, we can't use the Windows.UI.Xaml.Color type for doing property binding. We use Vector4 instead and this method ensures the conversion from color to Vector4 is done the same way by developers as is expected by generated code. 
+
+**Lottie feature support**
+* Handle opacity and color animated on the same element.
+
+**Code generation**
+* Rich generated comments about the Lottie contents, and the code generation. 
+    * We now generate comments that identify the version of the tool, when it was run, the input file, the command line arguments, etc, etc, so that it easier to keep track of how the file was generated, and to regenerate with identical options when the Lottie file changes.
```
Comment (derobinsMSFT) [inline VERSION_HISTORY.md:25]: >it [](start = 149, length = 2)

"it is" || "it's"

File: VERSION_HISTORY.md [hunk 5]
```diff
@@ -2,6 +2,64 @@
...
+    * Use "Visual.IsVisible" property for visibility control. This was previously not possible due to requirements to run downlevel (where this property was not available). 
+    * Use "CompositionShape.Scale" for visibility on shapes instead of TransformMatrix. This generates more efficient code and enables more optimizations. 
+    * Gradient stop optimization. After Effects does not remove redundant gradient stops, and simple gradients often have a redundant "middle" gradient stop. We can now detect and remove redundant gradient stops. 
+    * Eliminate redundant "Position" animations. 
+    * Compact expression strings to save some bytes. 
+    * Readability improvements for generated code. 
+    * Method naming improvements. 
+    * Matrices now have comments explaining what they do. 
+
+* C++/winrt
```
Comment (derobinsMSFT) [inline VERSION_HISTORY.md:45]: >winrt [](start = 6, length = 5)

"WinRT"?

File: VERSION_HISTORY.md [hunk 6]
```diff
@@ -2,6 +2,64 @@
...
+    * We now generate comments that identify the version of the tool, when it was run, the input file, the command line arguments, etc, etc, so that it easier to keep track of how the file was generated, and to regenerate with identical options when the Lottie file changes. 
+    * We generate various tables in the comments to help developers understand the contents of the generated file. These include information about markers, and counts of some important objects. Object counts are useful for determining whether a change to a Lottie file may have regressed performance. 
+
+* New code generation options.
+    * Added "-Namespace" option to put the generated code into a user's namespace. 
+    * Added "-Public" switch to generate public classes. Generated code defaults to internal, but can be overridden with this switch. 
+    * Added "-TestMode" switch to generate code that doesn't change given the same inputs (e.g. no timestamps are included in the output). Used for regression testing. 
+
+* Code quality improvements.
+    * Abstracted common patterns (e.g. creating a shape and setting its fill) into methods. This results in a smaller code that is easier to read.
```
Comment (derobinsMSFT) [inline VERSION_HISTORY.md:34]: >a smaller code [](start = 108, length = 14)

Wording-- how about "less code" or "a smaller file"?

File: VERSION_HISTORY.md (hunk 4 above)
Comment (derobinsMSFT) [inline VERSION_HISTORY.md:25]: >etc [](start = 131, length = 3)

Nit: period (etc.-- abbreviation). Same elsewhere in file where used.

File: VERSION_HISTORY.md [hunk 7]
```diff
@@ -2,6 +2,64 @@
...
+    * We generate various tables in the comments to help developers understand the contents of the generated file. These include information about markers, and counts of some important objects. Object counts are useful for determining whether a change to a Lottie file may have regressed performance. 
+
+* New code generation options.
+    * Added "-Namespace" option to put the generated code into a user's namespace. 
+    * Added "-Public" switch to generate public classes. Generated code defaults to internal, but can be overridden with this switch. 
+    * Added "-TestMode" switch to generate code that doesn't change given the same inputs (e.g. no timestamps are included in the output). Used for regression testing. 
+
+* Code quality improvements.
+    * Abstracted common patterns (e.g. creating a shape and setting its fill) into methods. This results in a smaller code that is easier to read. 
+    * Canonicalize progress remapping variables. These are the "t1", "t2", etc variables that are used to support spatial beziers. Canonicalizing can reduce the number of variables (and their associated costs).
```
Comment (derobinsMSFT) [inline VERSION_HISTORY.md:35]: >b [](start = 122, length = 1)

Capitalize.

File: VERSION_HISTORY.md [hunk 8]
```diff
@@ -2,6 +2,64 @@
...
+    * Compact expression strings to save some bytes. 
+    * Readability improvements for generated code. 
+    * Method naming improvements. 
+    * Matrices now have comments explaining what they do. 
+
+* C++/winrt
+    * Limited support for C++/winrt. This is enough support to satisfy the needs of some Microsoft teams, but should not be considered general support yet. Bug reports are welcome though and we will work towards full support. 
+
+**Metadata handling**
+* Added a general metadata facility for all CompositionObjects. This allows arbitrary data to be passed through the translator. It is used for Lottie metadata (e.g. markers), property binding, and object names, and object descriptions.
```
Comment (derobinsMSFT) [inline VERSION_HISTORY.md:49]: >and [](start = 193, length = 3)

Remove.

File: VERSION_HISTORY.md [hunk 9]
```diff
@@ -2,6 +2,64 @@
...
+
+**Parser performance**
+* Removed dependency on the Newtonsoft JSON parser. We now use the System.Text.Json parser. Parsing performance improved ~50%.
+* Parsing is more resilient to malformed and invalid Lottie files.
+
+**Lottie Viewer**
+* Remove support for code generation from the Lottie Viewer tool. Code generation can now only be done by the LottieGen.exe tool. This is so we can add lots of options to the code generator without complicating the Lottie Viewer UI. It also means there is only one way to generate code, which allows for more repeatable code generation in production environments.
+
+**Bug fixes**
+* Prevent use of PathKeyFrameAnimations before UAP v11. PathKeyFrameAnimationsare unreliable on earlier versions and may result in a crash.
```
Comment (derobinsMSFT) [inline VERSION_HISTORY.md:59]: >are [](start = 78, length = 3)

Space.

File: VERSION_HISTORY.md (hunk 1 above)
Comment (derobinsMSFT) [inline VERSION_HISTORY.md:14]: >  [](start = 159, length = 1)

Extra space?

File: VERSION_HISTORY.md (hunk 6 above)
Comment (simeoncran) [reply VERSION_HISTORY.md:34]: I think you're reacting to the "a" that shouldn't be there. I'll remove that. It really is supposed to say "smaller code".

---
In reply to: [427684023](https://github.com/windows-toolkit/Lottie-Windows/pull/261#discussion_r427684023) [](ancestors = 427684023)

File: VERSION_HISTORY.md (hunk 7 above)
Comment (simeoncran) [reply VERSION_HISTORY.md:35]: That surprised me - didn't realize Bezier was usually capitalized. Looks weird as "spatial Beziers", but ok.

---
In reply to: [427684771](https://github.com/windows-toolkit/Lottie-Windows/pull/261#discussion_r427684771) [](ancestors = 427684771)

File: VERSION_HISTORY.md [hunk 10]
```diff
@@ -2,6 +2,64 @@
...
+    * We generate various tables in the comments to help developers understand the contents of the generated file. These include information about markers, and counts of some important objects. Object counts are useful for determining whether a change to a Lottie file may have regressed performance.
+
+* New code generation options.
+    * Added "-Namespace" option to put the generated code into a user's namespace.
+    * Added "-Public" switch to generate public classes. Generated code defaults to internal, but can be overridden with this switch.
+    * Added "-TestMode" switch to generate code that doesn't change given the same inputs (e.g. no timestamps are included in the output). Used for regression testing.
+
+* Code quality improvements.
+    * Abstracted common patterns (e.g. creating a shape and setting its fill) into methods. This results in smaller code that is easier to read.
+    * Canonicalize progress remapping variables. These are the "t1", "t2", etc variables that are used to support spatial Beziers. Canonicalizing can reduce the number of variables (and their associated costs).
```
Comment (derobinsMSFT) [inline VERSION_HISTORY.md:35]: >etc [](start = 75, length = 3)

Period.

File: VERSION_HISTORY.md [hunk 11]
```diff
@@ -2,6 +2,64 @@
...
+    * Use "Visual.IsVisible" property for visibility control. This was previously not possible due to requirements to run downlevel (where this property was not available).
```
Comment (derobinsMSFT) [inline VERSION_HISTORY.md:36]: >  [](start = 172, length = 1)

Extra space.

**Comment by simeoncran** [comment]
Just FYI, this PR will never be completed - it's just here to get reviewed and to force a build. The contents that you are reviewing will get moved into the final release branch and into master though, so the review is still needed.

---
In reply to: [414927309](https://github.com/windows-toolkit/Lottie-Windows/pull/261#issuecomment-414927309) [](ancestors = 414927309)

**Pull request closed without merging**